src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -12464,21 +12517,29 @@ public SimpleGenerator()
             _sourceToAdd = string.Empty;
         }
 
-        public SimpleGenerator(string sourceToAdd)
+        public SimpleGenerator(string sourceToAdd, string fileName)
         {
             _sourceToAdd = sourceToAdd;
+            _fileName = fileName;
         }
 
         public void Execute(SourceGeneratorContext context)
         {
             if (!string.IsNullOrWhiteSpace(_sourceToAdd))
             {
-                context.AddSource("addedSource.cs", SourceText.From(_sourceToAdd, Encoding.UTF8));
+                context.AddSource(_fileName, SourceText.From(_sourceToAdd, Encoding.UTF8));
             }
         }
 
         public void Initialize(InitializationContext context)
         {
         }
     }
+
+    internal class SimpleGenerator2 : SimpleGenerator
```

Comment:
Can you explain why we need this class here? It doesn't seem to add anything over `SimpleGenerator`